### PR TITLE
fix: implement v-model on CNumberInput component

### LINKED ISSUE #150: Implement v-model on CNumberInput component
**Is your feature request related to a problem? Please describe.**
The number input doesn't have a 2 way binding implmentation.

**Describe the solution you'd like**
Implement v-model on CNumberInput

### PR DESCRIPTION
**Pull request by peoray**
Implement v-model on the component

## Description
- Added a model option
- Set the event to emit or update the props

## Motivation and Context

This allows the local state to keep track of changes on the component

fixes #150  

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/chakra-ui-core/src/CNumberInput/CNumberInput.js [hunk 1]
```diff
@@ -436,7 +444,11 @@ const CNumberInput = {
         align: 'stretch',
         w: this.isFullWidth ? 'full' : null,
         pos: 'relative'
-      }
+      },
+      nativeOn: {
+        input: this.emitValue
+      },
+      ref: 'input'
```
Comment (codebender828) [inline packages/chakra-ui-core/src/CNumberInput/CNumberInput.js:451]: Hey @peoray ! Thanks for the PR. Have a few things to make some changes for.

Even though the event there is the native input coming from the `CNumberInput` element, it's actually not the target element associated with the event. So I don't think it's a good idea to listen to the native event here and rely on it. 

The better place to handle the emitted changes would be inside the `this.handleChange` method. We only need to emit the `change` event whenever a new value is entered into the `CNumberInputField`.

So technically the `v-model` binding should work by changing the `model` property to be:
```js
model: {
  event: 'change',
  prop: 'value'
}
```

File: packages/chakra-ui-core/src/CNumberInput/CNumberInput.js [hunk 2]
```diff
@@ -426,6 +430,10 @@ const CNumberInput = {
     handleChange (event, value) {
       this.updateValue(event.target.value)
       this.$emit('change', event, value)
+    },
+    emitValue (event) {
+      this.$emit('input', event.target.value, event)
+      this.$emit('change', event)
```
Comment (codebender828) [inline packages/chakra-ui-core/src/CNumberInput/CNumberInput.js:436]: See comment on the `naviteOn` property.

File: packages/chakra-ui-core/src/CNumberInput/CNumberInput.js [hunk 3]
```diff
@@ -15,6 +15,10 @@ import CIcon from '../CIcon'
  */
 const CNumberInput = {
   name: 'CNumberInput',
+  model: {
+    prop: 'value',
+    event: 'input'
```
Comment (codebender828) [inline packages/chakra-ui-core/src/CNumberInput/CNumberInput.js:20]: See comment on `nativeOn` property.

**Comment by peoray** [comment]
@codebender828 Made the required changes.

**Comment by codebender828** [comment]
Thank you for working on this @peoray

**Comment by codebender828** [comment]
@all-contributors please add @peoray for code

**Pull request merged**